File: assign01/tcpclient.py
##########################
# Lab : Network Lab      #
# Name : Abhinav Kumar   #
# Enroll no.: 510817075  #
# Assignment : 01(Q.01)  #
##########################
import socket


def tcpclient(host=socket.gethostname(), port=9002):
    # created client endpoint
    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    client_socket.connect((host, port))  # request connection from server
    message = client_socket.recv(1024).decode('ascii')  # gets response
    print(message)  # prints response to terminal
    client_socket.close()  # closed client endpoint


if __name__ == "__main__":
    # if i want to connect to remote tcp server then i have to
    # pass its ip and port as argument of function below by default it will connect to 127.0.0.1
    tcpclient()

File: assign01/tcpclient.c
/********************
Lab : Network Lab
Name : Abhinav Kumar
Enroll no.: 510817075
Assignment : 01(Q.01)
*********************/
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <sys/socket.h>
#include <netinet/in.h>
int main()
{
    int network_socket;
    network_socket = socket(AF_INET, SOCK_STREAM, 0); //one end point sock_stream for tcp
    struct sockaddr_in server_address;                //creates server address
    server_address.sin_family = AF_INET;
    server_address.sin_port = htons(9002);       //htons convert int to port data type
    server_address.sin_addr.s_addr = INADDR_ANY; //assingns address to server
    int connection_status = connect(network_socket, (struct sockaddr *)&server_address, sizeof(server_address));
    if (connection_status == -1)
    {
        printf("Connection error");
    }
    printf("Request sent to server \n");
    char recv_data[250];
    recv(network_socket, &recv_data, sizeof(recv_data), 0); //flag is set to 0
    printf("Server's response :%s\n", recv_data);
    close(network_socket); //closes the endpoint so other process can use it
    return 0;
}
